include/boost/winasio/winhttp/winhttp_asio.hpp

```diff
@@ -135,6 +138,7 @@ void __stdcall BasicAsioAsyncCallback(HINTERNET hInternet, DWORD_PTR dwContext,
     BOOST_LOG_TRIVIAL(debug) << "winhttp callback error: " << err;
     ec.assign(pAR->dwError, boost::asio::error::get_system_category());
     // TODO: invoke handler
+    cpContext->on_send_request_failed.complete(ec, 0);
```

Comment:
you need to switch on ctx_state_type and invoke the handlers respectively.

Reply:
Will add 👍